src/lib/request.js

```diff
@@ -0,0 +1,10 @@
+export default function request(method, path, data = null) {
+    method = method.toUpperCase();
+
+    const request = new XMLHttpRequest();
+    request.open(method, path, true);
+    if (method === 'POST') {
+        request.setRequestHeader('Content-Type', 'application/json');
+    }
+    request.send(data);
```

Comment:
We should return the request object in case callers want to hook into handlers like onload, onerror, onprogress, etc.